pkgs/servers/mail/sympa/default.nix

```diff
@@ -0,0 +1,116 @@
+{ stdenv, perlPackages, fetchurl, makeWrapper
+, dataDir ? "/var/lib/sympa"
+}:
+
+let
+  pname = "sympa";
+  version = "6.2.44";
+in
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "https://github.com/sympa-community/${pname}/releases/download/${version}/${pname}-${version}.tar.gz";
+    sha256 = "10qzzjpl9vkkh3qlw4z44wmgcf8qlnazncl33xv4fv6wdg2wns2v";
+  };
+
+  configureFlags = [
+    "--without-initdir"
+    "--without-unitsdir"
+    "--without-smrshdir"
+
+    "--with-lockdir=/run/lock"
+    "--with-piddir=/run/sympa"
+    "--with-confdir=${dataDir}/etc"
+    "--sysconfdir=${dataDir}/etc"
+    "--with-spooldir=${dataDir}/spool"
+    "--with-expldir=${dataDir}/list_data"
+  ];
+  buildInputs = [ perlPackages.perl makeWrapper ];
+  propagatedBuildInputs = with perlPackages; [
+    ArchiveZip
+    CGI
+    CGIFast
+    ClassSingleton
+    DateTime
+    DBI
+    DateTimeFormatMail
+    DateTimeTimeZone
+    DigestMD5
+    Encode
+    FCGI
+    FileCopyRecursive
+    FileNFSLock
+    FilePath
+    HTMLParser
+    HTMLFormatter
+    HTMLTree
+    HTMLStripScriptsParser
+    IO
+    IOStringy
+    LWP
+    libintl_perl
+
+    MHonArc
+    MIMEBase64
+    MIMECharset
+    MIMETools
+    MIMEEncWords
+    MIMELiteHTML
+    MailTools
+    NetCIDR
+    ScalarListUtils
+    SysSyslog
+    TermProgressBar
+    TemplateToolkit
+    URI
+    UnicodeLineBreak
+    XMLLibXML
+
+    ### Features
+    Clone
+    CryptEksblowfish
+
+    DBDPg
+    DBDSQLite
+    DBDmysql
+
+    DataPassword
+    EncodeLocale
+    IOSocketSSL
+    MailDKIM
+    NetDNS
+    NetLDAP
+    NetSMTP
+    SOAPLite
+  ];
+
+  preInstall = ''
+    mkdir "$TMP/bin"
+    for i in chown chgrp chmod; do
+      echo '#!${stdenv.shell}' >> "$TMP/bin/$i"
+      chmod +x "$TMP/bin/$i"
+      PATH="$TMP/bin:$PATH"
```

Comment:
Move this one line lower. Also can't you just omit that?

Reply:
Oops. Omitting causes the build to fail in install phase:
```
make  install-exec-hook
make[4]: Entering directory '/build/sympa-6.2.44/src/cgi'
chown sympa /nix/store/2yv66g71p9bmy7i1gxry4gvp25wl203y-sympa-6.2.44/bin/wwsympa-wrapper.fcgi
chown: invalid user: 'sympa'
make[4]: [Makefile:783: install-exec-hook] Error 1 (ignored)
chgrp sympa /nix/store/2yv66g71p9bmy7i1gxry4gvp25wl203y-sympa-6.2.44/bin/wwsympa-wrapper.fcgi
chgrp: invalid group: 'sympa'
make[4]: [Makefile:784: install-exec-hook] Error 1 (ignored)
chmod 6755 /nix/store/2yv66g71p9bmy7i1gxry4gvp25wl203y-sympa-6.2.44/bin/wwsympa-wrapper.fcgi
chmod: changing permissions of '/nix/store/2yv66g71p9bmy7i1gxry4gvp25wl203y-sympa-6.2.44/bin/wwsympa-wrapper.fcgi': Operation not permitted
```